agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/localstorage/LocalFileLoader.java

```diff
@@ -41,6 +41,8 @@
   private static final OperationLogger operationLogger =
       new OperationLogger(LocalFileLoader.class, "Loading telemetry from disk");
 
+  @Nullable private File toBeDeletedFile;
```

Comment:
I think the logic would be easier to follow without this "global" state, by returning it from `loadTelemetriesFromDisk`, and then passing it back in to `updateProcessedFileStatus`

Reply:
loadTelemetriesFromDisk is returning ByteBuffer.  are you suggesting to return a struct?

Reply:
👍